c++/src/kj/async-inl.h

```diff
@@ -1084,7 +1084,7 @@ inline PromiseForResult<Func, void> evalNow(Func&& func) {
 template <typename Func>
 struct RetryOnDisconnect_ {
   static inline PromiseForResult<Func, void> apply(Func&& func) {
-    auto promise = evalNow(func);
+    auto promise = evalLater(func);
```

Comment:
Oh, the problem is probably that `evalLater()` here is capturing `func` by lvalue reference, so by the time it is executed, the reference is invalid.

You probably can't just copy `func` -- many lambdas are not copyable. I think you will have to wrap the entire body of `apply()` into an `evalLater`, rather than just change `evalNow` to `evalLater`.

Reply:
Wait, my changes aren't showing here... I must have messed up the push, since I already handled this issue locally (as mentioned in my comment 27 minutes ago). Let's try this again.

Reply:
To clarify, `evalLater()` and many other KJ APIs that accept functions use a convention that if you give it an rvalue, it takes ownership of that rvalue by move, but if you give it an lvalue, it takes a reference, on the assumption that you're naming a long-lived object that you want to call.

But neither one of those would be correct here.